src/custom-providers/azext_custom_providers/tests/latest/test_custom-providers_scenario.py

```diff
@@ -0,0 +1,82 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from azure.cli.testsdk import (ScenarioTest, ResourceGroupPreparer)
+
+
+class CustomProvidersScenarioTest(ScenarioTest):
+
+    @ResourceGroupPreparer(name_prefix='cli_test_custom_providers')
+    def test_custom_providers_common_scenario(self):
+
+        rp_name = self.create_random_name('clitest-crp', 20)
+        self.kwargs.update({
+            'rp_name': rp_name
+        })
+
+        self.cmd('az custom-providers resource-provider create '
+                 '--resource-group {rg} '
+                 '--name {rp_name} '
+                 '--location westus2 '
+                 '--action '
+                 'name=ping '
+                 'endpoint=https://ayniadjso4lay.azurewebsites.net/api '
+                 'routing_type=Proxy '
```

Comment:
well. I don't like `routing_type` as a key. Should it be `routing-type` at least

Reply:
I will keep use routing_type because:
1. It's not exposed as an argument, but just an key inside --action, just like key in --tags, so there is no convention it should use - instead of _
2. In actions.py, I defined a generic constructor "model(**kwargs)" which will map keyword in SDK model, if we use routing-type, then we need do a trick conversion from "-" to "_"

---
In reply to: [422454886](https://github.com/Azure/azure-cli-extensions/pull/1627#discussion_r422454886) [](ancestors = 422454886)